### PR TITLE
Pal 999 keep common library

### PR DESCRIPTION
**Pull request by dev930018**

### CODE REVIEW

**Review by pd104923 — DISMISSED** [review]

File: performance/pom.xml [hunk 1]
```diff
@@ -48,7 +48,7 @@
         <dependency>
             <groupId>uk.gov.gchq.data-gen</groupId>
             <artifactId>synthetic-data-generator</artifactId>
-            <version>0.0.2</version>
+            <version>0.0.3</version>
```
Comment (ac74475) [inline performance/pom.xml:51]: update to 0.0.4

File: example-library/pom.xml [hunk 2]
```diff
@@ -30,24 +30,45 @@
...
+            <artifactId>user-service</artifactId>
+            <version>0.5.0-${services.revision}</version>
+            <scope>provided</scope>
         </dependency>
         <!-- Dependency used to generate example data-->
         <dependency>
             <groupId>uk.gov.gchq.data-gen</groupId>
             <artifactId>synthetic-data-generator</artifactId>
-            <version>0.0.2</version>
+            <version>0.0.3</version>
```
Comment (ac74475) [inline example-library/pom.xml:50]: update to 0.0.4

Comment (pd104923) [reply example-library/pom.xml]: Updated

File: Jenkinsfile [hunk 3]
```diff
@@ -14,7 +14,7 @@
  * limitations under the License.
  */
 
-@Library('jenkinsfile-lib')_
+@Library('jenkinsfile-lib@feature/PAL-999-fix-examples-script')_
```
Comment (dev930018) [inline Jenkinsfile:17]: Remember to revert

File: performance/pom.xml [hunk 4]
```diff
@@ -48,7 +48,7 @@
         <dependency>
             <groupId>uk.gov.gchq.data-gen</groupId>
             <artifactId>synthetic-data-generator</artifactId>
-            <version>0.0.2</version>
+            <version>0.0.4</version>
```
Comment (dev930018) [inline performance/pom.xml:51]: Parameterise version

Comment (pd104923) [reply performance/pom.xml]: Parameterised this version

**Review by pd104923 — APPROVED** [review]

File: Jenkinsfile (hunk 3 above)
Comment (pd104923) [reply Jenkinsfile]: Reverted

File: example-library/src/test/java/uk/gov/gchq/palisade/example/library/ExampleUserServiceTest.java [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    @Autowired
+    UserConfiguration userConfiguration;
+
+    @Test
+    void testContextLoads() {
+        assertThat(userConfiguration).isInstanceOf(ExampleUserConfiguration.class);
+        UserPrepopulationFactory prepopulationFactory = userConfiguration.getUsers().get(0);
+        assertThat(prepopulationFactory).isInstanceOf(ExampleUserPrepopulationFactory.class);
+        User user = prepopulationFactory.build();
+        assertThat(user).isInstanceOf(ExampleUser.class);
```
Comment (developer01189998819991197253) [inline example-library/src/test/java/uk/gov/gchq/palisade/example/library/ExampleUserServiceTest.java:48]: add some .as("") to these tests

File: example-library/src/test/java/uk/gov/gchq/palisade/example/library/common/ExampleUserTest.java [hunk 6]
```diff
@@ -16,22 +16,26 @@
...
         User user = new ExampleUser().trainingCompleted(TrainingCourse.PAYROLL_TRAINING_COURSE).userId("bob").roles(Role.HR.name(), "another_role").auths("authorised_person", "more_authorisations");
 
         //when
-        byte[] bytesSerialised = JSONSerialiser.serialise(user, true);
-        User newUser = JSONSerialiser.deserialise(bytesSerialised, User.class);
+        byte[] bytesSerialised = MAPPER.writeValueAsBytes(user);
+        User newUser = MAPPER.readValue(bytesSerialised, ExampleUser.class);
 
         //then
         assertEquals("Deserialised user should be same class as original user", user.getClass(), newUser.getClass());
```
Comment (developer01189998819991197253) [inline example-library/src/test/java/uk/gov/gchq/palisade/example/library/common/ExampleUserTest.java:41]: update to use assertJ

File: pom.xml [hunk 7]
```diff
@@ -22,7 +22,7 @@
     <parent>
         <groupId>org.springframework.boot</groupId>
         <artifactId>spring-boot-starter-parent</artifactId>
-        <version>2.2.0.RELEASE</version>
+        <version>2.3.1.RELEASE</version>
```
Comment (developer01189998819991197253) [inline pom.xml:25]: why are we upgrading SpringBoot?

**Review by developer01189998819991197253 — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 7 above)
Comment (ac74475) [reply pom.xml:25]: to bring it in line with the services repo

File: example-library/src/test/java/uk/gov/gchq/palisade/example/library/ExampleUserServiceTest.java (hunk 5 above)
Comment (pd104923) [reply example-library/src/test/java/uk/gov/gchq/palisade/example/library/ExampleUserServiceTest.java:48]: Done

File: example-library/src/test/java/uk/gov/gchq/palisade/example/library/common/ExampleUserTest.java (hunk 6 above)
Comment (pd104923) [reply example-library/src/test/java/uk/gov/gchq/palisade/example/library/common/ExampleUserTest.java:41]: Done

**Review by developer01189998819991197253 — APPROVED** [review]

**Review by developer01189998819991197253 — APPROVED** [review]

**Pull request merged**